x-pack/plugins/reporting/export_types/csv_from_savedobject/server/execute_job.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { CONTENT_TYPE_CSV } from '../../../common/constants';
+// @ts-ignore
+import { createTaggedLogger, cryptoFactory, oncePerServer } from '../../../server/lib';
+import { JobDocPayload, KbnServer, Logger } from '../../../types';
+import {} from '../types';
+import { createGenerateCsv } from './lib/generate_csv';
+
+interface JobDocOutputPseudo {
+  content_type: 'text/csv';
+  content: string | null;
+}
+
+/*
+ * @return {Object}: pseudo-JobDocOutput. See interface JobDocOutput
+ */
+function executeJobFn(server: KbnServer) {
+  const crypto = cryptoFactory(server);
+  const config = server.config();
+  const serverBasePath = config.get('server.basePath');
+  const logger: Logger = {
+    debug: createTaggedLogger(server, ['reporting', 'savedobject-csv', 'debug']),
+    warning: createTaggedLogger(server, ['reporting', 'savedobject-csv', 'warning']),
+    error: createTaggedLogger(server, ['reporting', 'savedobject-csv', 'error']),
+  };
+
+  const generateCsv = createGenerateCsv(logger);
+  return async function executeJob(job: JobDocPayload): Promise<JobDocOutputPseudo> {
+    const { basePath, objects, headers: serializedEncryptedHeaders, jobParams } = job; // FIXME how to remove payload.objects for cleanup?
```

Comment:
Related to https://github.com/elastic/kibana/pull/31188/files#diff-413c6bc9d32af586a135bc6326d46cfbR109